// react-native-phonepe/PhonePeCheckout.js
import React from 'react';
import { NativeModules } from 'react-native';

class PhonePeCheckout extends React.Component
{
  /* Method to call by the mercahnts to initate the payment 
  Parameters are base64 body, checksum, endpoint, uiCallbackHandler */
  static launch (base64Body, checksum, endPoint, uiCallbackHandler)
  {
    NativeModules.PhonePeModule.getURL(getHostUrlcallback);
    function getHostUrlcallback(URL)
    {
        fetch(URL+endPoint,
          {
            method: "POST",
            body: JSON.stringify({
            request: base64Body
          }),
          headers: 
          {
            'Content-Type': 'application/json',
            'x-verify': checksum
          },
          })
          .then((response) => {
            if (response.ok) {
              return response.json();
            } else {
              alert("Something went wrong");
            }
          })
          .then((responseJson) => 
          {
            NativeModules.PhonePeModule.show(''+responseJson.data.redirectURL, uiCallbackHandler);
          })
          .catch((error) => 
          {
            console.error("Error: "+error);
          }); 
    }
  }
}
export default PhonePeCheckout;
